### ISSUE TITLE
Document that an access request must be made to download products from the USGS M2M API 

### ISSUE THREAD

**Issue by maximlt** [CS-SI/eodag#243]
```
usgs.USGSError: AUTH_UNAUTHROIZED: Your account does not have access to this endpoint
```

**Comment by sbrunato** [comment]
I do not have the same error. On my side, search is ok, but download hangs and fails with a `ConnectionError`. 

See https://github.com/kapadia/usgs/issues/57

**Comment by maximlt** [comment]
I think searching was OK for me too, I'll run it again and provide the complete traceback here if different from yours.

**Comment by maximlt** [comment]
This snippet:
```python
from eodag import EODataAccessGateway
from eodag.utils.logging import setup_logging
setup_logging(verbose=3)
dag = EODataAccessGateway('myconf.yml')
dag.set_preferred_provider("usgs")
search_criteria = dict(
    productType='L8_OLI_TIRS_C1L1',
    start='2017-03-01',
    end='2017-03-15',
    geom=[50, 50, 50.3, 50.3],
    items_per_page=30,
)
products, estimate = dag.search(**search_criteria)
print(f'Got {len(products)} products.')
dag.download(products[0])
```

led to this traceback:
```
Traceback (most recent call last):
  File "/home/maxime/TRAVAIL/06_EODAG/01_eodag/eodag/snippet_usgs.py", line 15, in <module>
    dag.download(products[0])
  File "/home/maxime/TRAVAIL/06_EODAG/01_eodag/eodag/eodag/api/core.py", line 1330, in download
    path = product.download(
  File "/home/maxime/TRAVAIL/06_EODAG/01_eodag/eodag/eodag/api/product/_product.py", line 257, in download
    fs_location = self.downloader.download(
  File "/home/maxime/TRAVAIL/06_EODAG/01_eodag/eodag/eodag/plugins/apis/usgs.py", line 184, in download
    download_options = api.download_options(
  File "/home/maxime/.pyenv/versions/eodag38/lib/python3.8/site-packages/usgs/api.py", line 77, in download_options
    _check_for_usgs_error(response)
  File "/home/maxime/.pyenv/versions/eodag38/lib/python3.8/site-packages/usgs/api.py", line 35, in _check_for_usgs_error
    raise USGSError('%s: %s' % (error_code, error))
usgs.USGSError: AUTH_UNAUTHROIZED: Your account does not have access to this endpoint
```

**Comment by maximlt** [comment]
OK I found on this [GitHub repo](https://github.com/MrChebur/usgs-machine-to-machine-API) that one actually has to request access to the USGS Machine-to-Machine API.

The link to request access is [this one](https://ers.cr.usgs.gov/profile/access). I filed a request and got quickly granted access to the API, for one year apparently.

After that, running the above snippet worked correctly, as long as the USGS end-to-end test.

I guess we should document that.
